cmd/syncthing/support_bundle.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2018 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package main
+
+import (
+	"archive/zip"
+	"io"
+
+	"github.com/syncthing/syncthing/lib/config"
+)
+
+// GetRedactedConfig redacting some parts of config
+func getRedactedConfig(s *apiService) config.Configuration {
```

Comment:
The definition of what's redacted is a bit wishy washy.
For example, if I don't want to reveal the devices I want to be connected to, perhaps we should replace device IDs with mock ones, and replace the content in the logs?

Reply:
Would not it kill the purpose of this feature request? if we mock the Device ID or anonymise the IP addresses at the logs here, it will be hard to troubleshoot an ongoing issue, which in this case we want this Support Bundle as helpful as possible?

Reply:
I don't think we need identities to debug things?

Reply:
loud thinking here; What about instead mocking the DeviceID, just showing the first 7 digits of it like it is seen at the UI?  If we mock whole DeviceID, it will be difficult to pinpoint the issue or correlate this with the DeviceID we have at the GUI. Imagine there are 100 Devices there..

Reply:
Sure, yet I suspect it will be easier to replace it with a integer based dummy than be able to serialise only first few characters.